pkg/result/processors/autogenerated_exclude.go

```diff
@@ -56,66 +63,96 @@ func (p *AutogeneratedExclude) shouldPassIssue(issue *result.Issue) (bool, error
 		return true, nil
 	}
 
-	if isSpecialAutogeneratedFile(issue.FilePath()) {
+	if !isGoFile(issue.FilePath()) {
 		return false, nil
 	}
 
-	fs, err := p.getOrCreateFileSummary(issue)
-	if err != nil {
-		return false, err
+	// The file is already known.
+	fs := p.fileSummaryCache[issue.FilePath()]
+	if fs != nil {
+		return !fs.generated, nil
 	}
 
+	fs = &fileSummary{}
+	p.fileSummaryCache[issue.FilePath()] = fs
+
+	if issue.FilePath() == "" {
+		return false, errors.New("no file path for issue")
+	}
+
+	if p.strict {
+		var err error
+		fs.generated, err = p.isGeneratedFileStrict(issue.FilePath())
+		if err != nil {
+			return false, fmt.Errorf("failed to get doc of file %s: %w", issue.FilePath(), err)
+		}
+	} else {
+		doc, err := getComments(issue.FilePath())
+		if err != nil {
+			return false, fmt.Errorf("failed to get doc of file %s: %w", issue.FilePath(), err)
```

Comment:
To simplify future debugging, we should use different errors for strict and non-strict paths.
Currently, if either `isGeneratedFileStrict` or `getComments` fails, the error returned is `fmt.Errorf("failed to get doc of file %s: %w", issue.FilePath(), err)`.